src/coral/experiment_tracking/extracking_setup.py

```diff
@@ -0,0 +1,37 @@
+"""This module contains the base class for an experiment tracking setup."""
+
+from abc import ABC, abstractmethod
+from collections.abc import Callable
+from dataclasses import dataclass
+from typing import Type, TypeAlias
+
+from omegaconf import DictConfig
+
+class ExTrackingSetup(ABC):
+    """Base class for an experiment tracking setup."""
+
+    @abstractmethod
+    def __init__(self, config: DictConfig) -> None:
+        """Initialise the experiment tracking setup.
+
+        Args:
+            config:
+                The configuration object.
+        """
+
+    @abstractmethod
+    def run_initialization(self) -> bool:
+        """Run the initialization of the experiment tracking setup.
+
+        Returns:
+            True if the initialization was successful, False otherwise.
```

Comment:
You mention here that it returns `False` if it wasn't successful (same with the `run_finalization` below), but you never return `False` with any of them. Perhaps you want to catch certain errors and return False if they appear, or something like that?

Alternatively, you can simply have these return nothing, and then they'll just raise the errors if they appear.

Reply:
removed return type